### PR TITLE
ob thunk pack: relax excessive checks

### PR DESCRIPTION
**Pull request by alexfmpe**
This makes `ob thunk pack` succeed if 
  * all branches were pushed *somewhere* 
  * branch being thunked was pushed to specified upstream 

both apply, instead of the current behavior of failing unless all remotes have all branches, which can be a real annoyance.

### CODE REVIEW

File: lib/command/src/Obelisk/Command/Thunk.hs [hunk 1]
```diff
@@ -775,6 +778,17 @@ getThunkPtr' checkClean thunkDir upstream = do
         }
       }
 
+isBranchPushed :: MonadObelisk m => FilePath -> Text -> Text -> m Bool
+isBranchPushed thunkDir hd rm = do
+  hasBranch <- not . null . T.lines <$> readGitProcess thunkDir ["ls-remote", "--heads", T.unpack rm, T.unpack hd]
```
Comment (alexfmpe) [inline lib/command/src/Obelisk/Command/Thunk.hs]: This fixes an unfriendly error when a branch does not even exist in a remote, e.g.
```
RawCommand "nix-shell" ["-p","git","--run","'git' '-C' 'reflex-platform' 'merge-base' 'f4c17b91e6104830667ded50e950dd08f5e79bfe' 'remotes/origin/fix-ios-simulator'"] failed with exit code: 128
```

**Comment by ryantrinkle** [comment]
@alexfmpe Could we add some tests for this?  Especially the negative case, i.e. when things have not been pushed.  I don't want to delete data unexpectedly.

**Comment by Ericson2314** [comment]
@alexfmpe I think the status quo now subsumes this?

**Comment by alexfmpe** [comment]
@Ericson2314 I agree. Closing
